src/main/java/com/mongodb/jdbc/BsonTypeInfo.java

```diff
@@ -56,16 +57,18 @@ public enum BsonTypeInfo {
     BSON_DECIMAL("decimal", BsonType.DECIMAL128, Types.DECIMAL, false, 34, 34, 10, 34, 34, 16),
     BSON_MINKEY("minKey", BsonType.MIN_KEY, Types.OTHER, false, 0, 0, 0, null, null, null),
     BSON_MAXKEY("maxKey", BsonType.MAX_KEY, Types.OTHER, false, 0, 0, 0, null, null, null),
-    BSON_BSON("bson", BsonType.UNDEFINED, Types.OTHER, false, 0, 0, 0, null, null, null);
+    BSON_BSON("bson", BsonType.UNDEFINED, Types.OTHER, false, 0, 0, 0, null, null, null),
+    BSON_DOCUMENT("document", BsonType.DOCUMENT, Types.OTHER, false, 0, 0, 0, null, null, null);
```

Comment:
We already have the `BSON_OBJECT` variant which represents the same thing.

The problem is we have two words for the same concept 😞 The [docs](https://www.mongodb.com/docs/manual/reference/bson-types/) list "object" as the BSON type name, whereas the Java (and, I think, all other) driver(s) use the term "document". Given the extensive use of `BSON_OBJECT` throughout other files in the JDBC driver, I'd say we should defer to using that one instead of this new one. However, it is also an option to update the name of `BSON_OBJECT` to `BSON_DOCUMENT` if we prefer the term "document" to "object".

Reply:
I had questions about this when I was making the addition. I think for our uses, having a concrete `BsonDocument` converted to a `BsonTypeInfo.BSON_OBJECT` would be okay. Thoughts?

Reply:
Agreed, it will be okay! It's the same type and the `BsonTypeInfo` we've already used in all other classes 👍